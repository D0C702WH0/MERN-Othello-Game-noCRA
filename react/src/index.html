<!DOCTYPE html>
<html lang="fr">
  <head>
    <title>zoOthello</title>
    <base href="/"/>
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="stylesheet" href="style.css"/>
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
    integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous" />
    <link href="https://fonts.googleapis.com/css?family=Patua+One&display=swap" rel="stylesheet"/>    
    <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet"/>
  </head>
  <body>
    <div id="root"></div>
  </body>
</html>