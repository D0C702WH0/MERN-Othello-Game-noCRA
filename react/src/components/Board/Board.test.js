import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Board from './Board'
import reversi from 'reversi'

const GameGameInstance = reversi.Game
const game = new GameGameInstance()

let wrapper
const handleClick = jest.fn()
beforeEach(() => {
  wrapper = render(
    <Board board={game._board} click={handleClick}/>
  )
})

describe('Board', () => {
  it('Basic rendering', () => {
    const { asFragment } = wrapper
    expect(asFragment()).toMatchSnapshot()
  })

  it('Board has 8 Rows', () => {
    const { queryAllByTestId } = wrapper
    const rows = queryAllByTestId('row')
    expect(rows).toHaveLength(8)
  })

  it('Board has 64 squares', () => {
    const { queryAllByTestId } = wrapper
    const squares = queryAllByTestId('square')
    expect(squares).toHaveLength(64)
  })

  it('Board has 2 black pawns', () => {
    const { queryAllByTestId } = wrapper
    const redDucks = queryAllByTestId('test-pawn-red')
    expect(redDucks).toHaveLength(2)
  })

  it('Board has 2 white pawns', () => {
    const { queryAllByTestId } = wrapper
    const yellowDucks = queryAllByTestId('test-pawn-yellow')
    expect(yellowDucks).toHaveLength(2)
  })
})
