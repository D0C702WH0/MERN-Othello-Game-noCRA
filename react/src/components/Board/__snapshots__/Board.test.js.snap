// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Basic rendering 1`] = `
<DocumentFragment>
  <div
    class="board_container"
  >
    <div
      class="board"
    >
      <div
        class="row"
        data-testid="row"
      >
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
      </div>
      <div
        class="row"
        data-testid="row"
      >
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
      </div>
      <div
        class="row"
        data-testid="row"
      >
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
      </div>
      <div
        class="row"
        data-testid="row"
      >
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          >
            <img
              alt="duck pawn"
              class="duckFace yellow"
              data-testid="test-pawn-yellow"
              src="./assets/img/duck-yellow.png"
            />
          </div>
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          >
            <img
              alt="duck pawn"
              class="duckFace red"
              data-testid="test-pawn-red"
              src="./assets/img/duck-red.png"
            />
          </div>
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
      </div>
      <div
        class="row"
        data-testid="row"
      >
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          >
            <img
              alt="duck pawn"
              class="duckFace red"
              data-testid="test-pawn-red"
              src="./assets/img/duck-red.png"
            />
          </div>
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          >
            <img
              alt="duck pawn"
              class="duckFace yellow"
              data-testid="test-pawn-yellow"
              src="./assets/img/duck-yellow.png"
            />
          </div>
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
      </div>
      <div
        class="row"
        data-testid="row"
      >
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
      </div>
      <div
        class="row"
        data-testid="row"
      >
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
      </div>
      <div
        class="row"
        data-testid="row"
      >
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
        <div
          class="zoom"
          tabindex="0"
        >
          <div
            class="square"
            data-testid="square"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
